clang/test/CIR/CodeGen/cast.cpp

```diff
@@ -76,6 +76,10 @@ int cStyleCasts_0(unsigned x1, int x2, float x3, short x4, double x5) {
   float bf = (float)ib; // bool to float
   // CHECK: %{{[0-9]+}} = cir.cast(bool_to_float, %{{[0-9]+}} : !cir.bool), f32
 
+  void* bpv = (void*)ib; // bool to pointer, which is done in two steps
+  // CHECK: %[[TMP:[0-9]+]] = cir.cast(bool_to_int,  %{{[0-9]+}} : !cir.bool), !u64i
+  // CHECK: %{{[0-9]+}} = cir.cast(int_to_ptr, %[[TMP]] : !u64i), !cir.ptr<!void>
```

Comment:
Alternatively we could define a new type of cast: bool_to_ptr, and expand that later during LowerPrepare. However, since there's no current use case for this, I'd say we keep as is for now.

Reply:
I started in that direction by creating `CastKind::bool_to_ptr`.  But when I saw that integer-to-pointer was already a two-step process and bool-to-pointer was already going through that code, it was easiest to tweak the existing code to correctly handle bool-to-pointer.

> since there's no current use case for this

That's an important point.  Bool-to-pointer conversions will never happen in real-world code.  We need this for completeness, but we shouldn't spend time worrying about doing it right.